src/plugins/vis_type_timeseries/public/application/components/vis_editor.tsx

```diff
@@ -212,21 +218,13 @@ export class VisEditor extends Component<TimeseriesEditorProps, TimeseriesEditor
     const dataStart = getDataStart();
 
     dataStart.indexPatterns.getDefault().then(async (index) => {
-      const defaultIndexTitle = index?.title ?? '';
-      const indexPatterns = extractIndexPatternValues(this.props.vis.params, defaultIndexTitle);
+      const indexPatterns = extractIndexPatternValues(this.props.vis.params, index);
       const visFields = await fetchFields(indexPatterns);
 
       this.setState((state) => ({
+        defaultIndex: index,
         model: {
           ...state.model,
```

Comment:
I think that now can be simplifed to:
```
      this.setState({
        defaultIndex: index,
        visFields,
      });
```
You don't change something to model, so no need to update it, if I understand the code correctly

Reply:
good point! thank you